frontend/src/host_orchestrator/orchestrator/userartifacts.go

```diff
@@ -168,6 +169,11 @@ func Untar(dst string, src string) error {
 		switch header.Typeflag {
 		case tar.TypeDir:
 			if _, err := os.Stat(target); err != nil {
+				oldmask := syscall.Umask(0)
+				defer func() {
```

Comment:
Seems defer keyword executes when right before function ends.

Reply:
Removed defer block and added code for restoring old mask